superduper-io/superduper
# Automate docs build

## Description

See #461

## Results

https://rec.github.io/superduperdb-stealth/

https://superduperdb.github.io/superduperdb-stealth/

*** NOTE NOTE NOTE NOTE: the URLs are PUBLIC.  I can remove this in a trice, it's to test it.

## How to test as a reviewer

* Patch in this commit (perhaps with `git new -d rec:docs-build`)
* `python -m superduperdb docs --help`
* `python -m superduperdb docs 'My first doc build' -o`

Now go to your fork on github, Settings/Pages, and enable `Deploy From A Branch`, then select `gh-pages` and `/root`, like this:

![Screenshot 2023-07-19 at 14 46 25](https://github.com/SuperDuperDB/superduperdb-stealth/assets/253362/faba8c02-3a53-4a3a-9e5c-099d0edb38fa)

In a few minutes, you should see the code at 

https://<YOUR-NAME-HERE>.github.io/superduperdb-stealth/

## CLI documentation

```
$ python -m superduperdb docs --help
                                                                                                                    
 Usage: python -m superduperdb docs [OPTIONS] [COMMIT_MESSAGE]                                                      
                                                                                                                    
 Build documentation, optionally committing and pushing got                                                         
                                                                                                                    
╭─ Arguments ──────────────────────────────────────────────────────────────────────────────────────────────────────╮
│   commit_message      [COMMIT_MESSAGE]  The git commit message for the docs update.An empty message means do not │
│                                         commit and push.                                                         │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Options ────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --open  -o        If true, open the index.html of the generated pages on completion                              │
│ --help  -h        Show this message and exit.                                                                    │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```
